### ISSUE TITLE
TypeError: Cannot read property 'toString' of null

### ISSUE THREAD

**Issue by twnaing** [vuex-orm/vuex-orm#304]
My Model has the following field definition

```js
  static fields () {
    return {
      id: this.attr(null),
      name: this.string(''),
      organization_id: this.attr(1)
    }
  }
```
I am using `vuex-orm` with `plugin-axios` (both setup as plugins in Nuxtjs).

With the following code

```js
Team.create({data: {name: 'tester', organization_id: 1}})
```

> TypeError: Cannot read property 'toString' of null

error arises.

related to #151 and the error arises from [vuex-orm.js:2773](https://github.com/vuex-orm/vuex-orm/blob/master/dist/vuex-orm.js#L2773) because `newId` is `null` at that point.

```js
        IdFixer.processRecords = function (records, query) {
            return Object.keys(records).reduce(function (newRecords, id) {
                var record = records[id];
                var newId = query.model.id(record);
                var newStringId = isNaN(newId) ? newId : newId.toString(); // error here
                if (newId === undefined || id === newStringId) {
                    newRecords[id] = record;
                    return newRecords;
                }
                newRecords[newStringId] = __assign({}, record, { $id: newId });
                return newRecords;
            }, {});
        };
        return IdFixer;
```

What am I doing wrong? or is it a bug?

**Comment by kiaking** [comment]
@twnaing Hi! Thank you for the report! Not sure what is happening. Did you register the model to the database? If yes, is it possible for you to create simple reproduction code for this? It might be a defect.

**Comment by lino** [comment]
The issue is that the default value of id is null and the IdFixer component is broken, as it accesses toString of null.

**Comment by lino** [comment]
The issue can be mitigated by changing primary keys to this.increment()

**Comment by kiaking** [comment]
Ah thank you for the report! Need to look into it. I'll mark this as a bug and try making test case first.

**Comment by lucasb** [comment]
Hi @kiaking! I have faced the same issue here. Let me explain what I found, maybe it can be helpful:
Model definition:
``` javascript
  static fields () {
    return {
      id: this.attr(null),
      name: this.string('')
    }
  }
```
when I call the new method the same error message rise up.
``` javascript
Model.new()
// TypeError: Cannot read property 'toString' of null
```
I believe the issue here is happening because of how ```isNaN``` works.
when the value is ```undefined``` or a ```String``` it returns true, but when it's ```null``` the return is false. Look here:
``` javascript
isNaN(undefined) // true
isNaN('undefined') // true
isNaN(null) // false
```
A possible fix is to guarantee that value is valid before check if it is Not a Number. See here:

``` javascript
 IdFixer.processRecords = function (records, query) {
     return Object.keys(records).reduce(function (newRecords, id) {
         var record = records[id];
         var newId = query.model.id(record);
         var newStringId = newId && isNaN(newId) ? newId : newId.toString(); // adding safe check here
         if (!newId || id === newStringId) { // changing to get undefined and null cases here.
             newRecords[id] = record;
             return newRecords;
         }
         newRecords[newStringId] = __assign({}, record, { $id: newId });
         return newRecords;
     }, {});
 };
 return IdFixer;
```

**Comment by kiaking** [comment]
Uhhhhh thank you so much for the advise! I didn't know that isNan issue... Thanks!

**Comment by kiaking** [comment]
OK I think I managed to fix this with #374. I'll be making a new release soon.

**Comment by kiaking** [comment]
This issue is fixed with Vuex ORM 0.31.12 🔥 
https://github.com/vuex-orm/vuex-orm/releases/tag/v0.31.12

Thank you so much for your help! 🎉
